framework_tvb/tvb/core/services/import_service.py

```diff
@@ -314,36 +293,112 @@ def _store_imported_images(self, project):
+                self._store_imported_datatypes_in_db(project, operation_datatypes, dt_burst_mappings,
+                                                     burst_ids_mapping)
+                imported_operations.append(operation_entity)
+            else:
+                view_model = self._get_main_view_model(vm_paths)
+
+                if view_model:
+                    start_date = datetime.now()
+                    alg = SimulatorAdapter().stored_adapter
+
+                    op = self.get_new_operation_for_view_model(project, view_model, alg.id)
+                    op.meta_data = '{"from": "Import"}'
+                    op.status = STATUS_FINISHED
+                    op.create_date = start_date
+                    op.start_date = start_date
+                    op.algorithm = alg
+                    op.visible = True
+                    op.completion_date = datetime.now()
+                    operation_entity = dao.store_entity(op)
+                    imported_operations.append(operation_entity)
+
+                # Store the DataTypes in db
+                if dt_paths:
+                    self._store_datatypes_from_path_in_db(dt_paths)
 
-            # Rename operation folder with the ID of the stored operation
-            new_operation_path = FilesHelper().get_operation_folder(project.name, operation_entity.id)
-            if old_operation_folder != new_operation_path:
-                # Delete folder of the new operation, otherwise move will fail
-                shutil.rmtree(new_operation_path)
-                shutil.move(old_operation_folder, new_operation_path)
+        return imported_operations
 
-            operation_datatypes = self._load_datatypes_from_operation_folder(new_operation_path, operation_entity,
-                                                                             datatype_group)
-            self._store_imported_datatypes_in_db(project, operation_datatypes, dt_burst_mappings, burst_ids_mapping)
-            imported_operations.append(operation_entity)
+    @staticmethod
+    def _get_operation_paths(paths):
+        return [f.path for f in os.scandir(paths) if f.is_dir()]
 
-        return imported_operations
+    def get_new_operation_for_view_model(self, project, view_model, alg_id):
+        op_param = '{"gid": "' + str(view_model.gid) + '"}'
+        op = Operation(project.fk_admin, project.id, alg_id, op_param)
+        return op
+
+    @staticmethod
+    def _get_view_model_and_datatypes_paths(import_path):
+        vm_paths = []
+        dt_paths = []
+        for root, _, files in os.walk(import_path):
+            for file in files:
+                if file.endswith(".h5"):
+                    vm_file_oath = root + "\\" + file
+                    h5_class = H5File.h5_class_from_file(vm_file_oath)
+                    if h5_class is ViewModelH5:
+                        vm_paths.append(vm_file_oath)
```

Comment:
We could check here whether the current ViewModel is the main one and return only that path. We do not use the other paths anywhere else anyway.